src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -20,6 +21,7 @@ internal class RemoteJSRuntime : JSRuntime
         private readonly CircuitOptions _options;
         private readonly ILogger<RemoteJSRuntime> _logger;
         private CircuitClientProxy _clientProxy;
+        private readonly ConcurrentDictionary<long, DotNetStreamReference> _pendingDotNetToJSStreams = new();
```

Comment:
Why do we need the concurrent dictionary here? Shouldn't this already be scoped to the circuit?

Reply:
[`cancellationToken.Register`](https://github.com/dotnet/aspnetcore/pull/34817/files#diff-7fac6aa482260bd9b68de9fd59db0da9d8532c59ecdfd4468c0f51d371b562aaR167-R174) which is used to evict streams which aren't read from after the timeout, [does _not_ capture the Synchronization context.](https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken.register?view=net-5.0). Hence, without a `ConcurrentDictionary` we could run into potential issues when the cancellation token expires (and attempts to evict the stream) at the exact time we get a call into `TryClaimPendingStreamForSending` to evict the stream for consumption.